src/session_handle.cpp

```diff
@@ -351,6 +351,16 @@ namespace libtorrent
 #endif
 	}
 
+	bool session_handle::set_dht_storage(dht::dht_storage_constructor_type sc)
```

Comment:
do you think it's important that this function returns a value? It's quite expensive, since the calling thread will have to be blocked waiting for the main thread. If it's possible to have it be an async. call, that would be a lot more efficient